frontend/src/metabase/routes.jsx

```diff
@@ -219,10 +218,35 @@ export const getRoutes = store => {
             <Route path="metabot" component={QueryBuilder} />
           </Route>
 
-          <Route path="browse" component={BrowseApp}>
-            <IndexRoute component={DatabaseBrowser} />
-            <Route path=":slug" component={SchemaBrowser} />
-            <Route path=":dbId/schema/:schemaName" component={TableBrowser} />
+          <Route path="browse">
+            <IndexRedirect to="/browse/models" />
+            <Route path="models" component={() => <BrowseApp tab="models" />} />
+            <Route
+              path="databases"
+              component={() => <BrowseApp tab="databases" />}
+            />
+            <Route
+              path="databases/:slug"
+              component={({ params }) => (
+                <BrowseApp tab="databases">
+                  <SchemaBrowser params={params} />
+                </BrowseApp>
+              )}
+            />
+            <Route
+              path="databases/:dbId/schema/:schemaName"
+              component={({ params }) => (
+                <BrowseApp tab="databases">
+                  <TableBrowser params={params} />
+                </BrowseApp>
+              )}
+            />
+
+            <Redirect from=":dbId-:slug" to="databases/:dbId-:slug" />
```

Comment:
I think a comment here would be helpful to let future developers know that this is to support legacy paths in versions 48 and earlier.